Sources/Core/EthereumNetwork/RequestParameter/APIRequestParameterType.swift

```diff
@@ -0,0 +1,28 @@
+//
+//  Web3+APIRequestParameter.swift
+//  Web3swift
+//
+//  Created by Yaroslav on 24.05.2022.
+//
+
+import Foundation
+
+/// Protocol to restrict supported types which can be passed into `RequestParameter` to a node.
+///
+/// Due to internal logic and swift itself restrictions, there's lack of encoding generic types
+/// so current implementation of `RequestParameter`s belongs on hardcoded supported types.
+///
+/// Please see `RequestParameter` documentation for more details.
+protocol APIRequestParameterType: Encodable { }
```

Comment:
What's the difference from `APIRequestParameterElementType`.

It looks like we can get rid of `APIRequestParameterElementType` and leave only `APIRequestParameterType`.



There should be no issues updating
```
extension Array: APIRequestParameterType where Element: APIRequestParameterElementType { }
```
to
```
extension Array: APIRequestParameterType where Element: APIRequestParameterType { }
```

Reply:
I'll be happy if it will work) on the development stage it didn't, so this is the only reason of second protocol happening.) but i'll dig shortly about your proposal, hope it'll working that way.

Reply:
It worked.